src/components/features/home/contact/contact-form/ContactForm.tsx

```diff
@@ -0,0 +1,79 @@
+import { Form } from 'react-router-dom'
+import './contact-form.scss'
+import { useForm } from 'react-hook-form'
+import * as yup from 'yup'
+import { yupResolver } from '@hookform/resolvers/yup'
+
+const schema = yup.object({
+  name: yup
+    .string()
+    .required('Proszę podać swoje imię.')
+    .matches(/^(?=.{1,50}$)[a-z]+(?:['_.\s][a-z]+)*$/i),
+  email: yup
+    .string()
+    .email('Proszę podać poprawny adres email.')
+    .required('Proszę podać swój adres email.'),
+  message: yup.string().required('Proszę podać swoją wiadomość.'),
+  agreeTerms: yup
+    .boolean()
+    .required('Proszę zaakceptować zgodę na przetwarzanie danych osobowych.')
+    .oneOf(
+      [true],
+      'Proszę zaakceptować zgodę na przetwarzanie danych osobowych.'
+    ),
+})
+
+type FormData = yup.InferType<typeof schema>
+
+const ContactForm = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormData>({
+    resolver: yupResolver(schema),
+  })
+  const onSubmit = (data: FormData) => console.log(data)
+
+  return (
+    <Form className="form" onSubmit={handleSubmit(onSubmit)} method="post">
+      <div className="form__container">
+        <input type="text" placeholder="Imię" {...register('name')} className={errors.name ? 'input__alert-active': ''} />
+        <div className='error__container'>
+        <span className="form__error">{errors.name?.message}</span>
+        </div>        <input type="email" placeholder="Email" {...register('email')} className={errors.email ? 'input__alert-active': ''}/>
+        <div className='error__container'>
+        <span className="form__error">{errors.email?.message}</span>
+        </div>
+        <textarea
+          placeholder="Wiadomość"
+          className={`input__message ${
+            errors.message ? 'textarea__alert-active' : ''
+          }`}
+          {...register('message')}
+          
+        />
+<div className='error__container'>
+        <span className="form__error">{errors.message?.message}</span>
+        </div>        <div className="checkbox__row">
+          <input
```

Comment:
Trzymając się BEM'a uważałabym, aby nazwy klas się nie zmieniały a odnosiły się do jednej konkretnej rzeczy, czyli tutaj **"contact-form"**. 

A więc żeby zachować spójność fajowsko jakby wszystkie elementy dotyczące tego tak się zaczynały (checkboxów, errorów i tym podobnych w projekcie może być dużo), a więc zamiast "input__checkbox" to "contact-form__input-checkbox", "contact-form__error-container".

Wiem że krótsze nazwy są ładniejsze ale w BEM'ie jest spoko to że jest bardziej opisowy i uporządkowany a do daje duży plus do wspólnej pracy nad projektem